### PR TITLE
OG-1009: Migrate GSN RelayProvider from Web3.js to Ethers.js

### PR DESCRIPTION
**Pull request by forshtat**
The Web3.js library is more than 1.1 MB in size minimised.

We should be able to replace it with Ethers.js that is ~200 kb.

Using Ethers.js v5 as GSN is a CommonJS package and v6 is an ESModule.

Resolves https://github.com/opengsn/gsn/issues/910

### CODE REVIEW

**Comment by forshtat** [comment]
CTF demo (https://github.com/opengsn/ctf-react) before and after migration:

Screenshot 2023-02-28 at 20 19 48

Screenshot 2023-02-28 at 20 19 37

File: packages/common/src/ContractInteractor.ts [hunk 1]
```diff
@@ -192,7 +191,7 @@ export class ContractInteractor {
     this.maxPageCount = maxPageCount ?? Number.MAX_SAFE_INTEGER
     this.logger = logger
     this.versionManager = versionManager ?? new VersionsManager(gsnRuntimeVersion, gsnRequiredVersion)
-    this.web3 = new Web3(provider as any)
+    // this.web3 = new Web3(provider as any)
```
Comment (shahafn) [inline packages/common/src/ContractInteractor.ts:194]: Please remove

File: packages/common/src/LightTruffleContract.ts [hunk 2]
```diff
@@ -49,21 +56,27 @@ function retype (outputs?: AbiOutput[], ret?: any): any {
 }
 
 export class Contract<T> {
-  web3!: Web3
+  // web3!: Web3
```
Comment (shahafn) [inline packages/common/src/LightTruffleContract.ts:59]: Remove

File: packages/common/src/Utils.ts [hunk 3]
```diff
@@ -113,36 +113,41 @@ export async function getEip712Signature<T extends MessageTypes> (
...
-        error = result.error as any
-      }
-      if ((errorAsBoolean(error)) || result == null) {
-        reject((error as any).message ?? error)
-      } else {
-        resolve(correctV(result.result))
-      }
-    })
-  })
+  // return await new Promise((resolve, reject) => {
```
Comment (shahafn) [inline packages/common/src/Utils.ts:116]: Remove

File: packages/common/src/Utils.ts [hunk 4]
```diff
@@ -113,36 +113,41 @@ export async function getEip712Signature<T extends MessageTypes> (
...
+    params: [senderAddress, dataToSign],
+    jsonrpc: '2.0',
+    id: Date.now()
+  }
+
+  // TODO: check promises logic is not broken here; apply fixes;
+  const signedStuff = await provider.send(paramBlock.method, paramBlock.params)
+
+  return signedStuff
+  // method.bind(web3.currentProvider)(paramBlock, (error: Error | string | null | boolean, result?: JsonRpcResult) => {
```
Comment (shahafn) [inline packages/common/src/Utils.ts:137]: Remove

**Review by shahafn — APPROVED** [review]

**Pull request merged**